pkg/kubelet/client/kubelet_client.go

```diff
@@ -113,6 +148,13 @@ func makeTransport(config *KubeletClientConfig, insecureSkipTLSVerify bool) (htt
 // transportConfig converts a client config to an appropriate transport config.
 func (c *KubeletClientConfig) transportConfig() *transport.Config {
 	cfg := &transport.Config{
+		// always bust the client-go TLS cache instead of only when KubeletClientConfig.Lookup is set.
+		// this allows us to safely mutate the underlying *http.Transport in mutateTransportToValidateNodeName.
+		// note that we do not need the client-go TLS cache since we make and re-use a small number of transports.
+		// TODO add unit test
+		//  1. unique transport per call for same inputs
+		//  2. unwrapping still works at high log levels when the transport is actually wrapped
+		Proxy: http.ProxyFromEnvironment,
```

Comment:
I don't love relying on this to get a unique transport without a test ... at least unit test this results in two distinct underlying http.Transport objects when set

edit: ha, just noticed you literally have a TODO to add a unit test for this

also, if you want to avoid trying to support HTTP_PROXY / HTTPS_PROXY in combination with this (since egress selector is available), you could set this to a no-op no-proxy function (just return nil always)

also, I think we only need to do this when config.TLSClientConfig.ValidateNodeName is set